models/classes/Translation/Service/ResourceMetadataPopulateService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\Translation\Service;
+
+use core_kernel_classes_Literal;
+use core_kernel_classes_Resource;
+use oat\generis\model\data\Ontology;
+use oat\tao\model\Translation\Entity\AbstractResource;
+
+class ResourceMetadataPopulateService
+{
+    private Ontology $ontology;
+    private array $metadata;
+
+    public function __construct(Ontology $ontology)
+    {
+        $this->ontology = $ontology;
+    }
+
+    public function addMetadata(string $resourceType, string $metadataUri): void
+    {
+        $this->metadata[$resourceType] = empty($this->metadata[$resourceType]) ? [] : $this->metadata[$resourceType];
```

Comment:
Can't we use here the following?
```suggestion
        $this->metadata[$resourceType] ??= [];
```